calendar-webadmin/src/main/java/com/linagora/calendar/webadmin/ResourceRoutes.java

```diff
@@ -156,84 +174,85 @@ private ResourceDTO getResource(Request req) {
-                .statusCode(HttpStatus.NOT_FOUND_404)
-                .type(ErrorResponder.ErrorType.NOT_FOUND)
-                .message("Resource do not exist")
-                .cause(new RuntimeException())
-                .haltError();
-        }
+        ResourceId resourceId = new ResourceId(req.params("id"));
+
+        return findNotDeletedResource(resourceId)
+            .flatMap(currentResource -> resourceAdministratorService.revokeAdmins(currentResource)
+                .then(resourceDAO.softDelete(resourceId)))
+            .doOnSuccess(any -> res.status(HttpStatus.NO_CONTENT_204))
+            .thenReturn(StringUtils.EMPTY)
+            .onErrorMap(ResourceNotFoundException.class, notFoundException -> {
+                LOGGER.warn("Failed to delete resource {}: not found or invalid.", resourceId.value());
+                return resourceNotFound(resourceId);
+            })
+            .block();
     }
 
     private String createResource(Request req, Response res) throws JsonExtractException {
-        ResourceCreationDTO dto = creationDTOJsonExtractor.parse(req.body());
-
-        OpenPaaSId creatorId = userDAO.retrieve(Username.of(dto.creator))
-            .switchIfEmpty(Mono.error(() -> new IllegalArgumentException("creator '" + dto.creator + "' must exist")))
-            .block().id();
-        OpenPaaSId domainId = domainDAO.retrieve(Domain.of(dto.domain))
-            .switchIfEmpty(Mono.error(() -> new IllegalArgumentException("domain '" + dto.domain + "' must exist")))
-            .block().id();
-        List<ResourceAdministrator> administrators = retrieveResourceAdministrators(dto.administrators);
-
-        ResourceId resourceId = resourceDAO.insert(new ResourceInsertRequest(administrators,
-            creatorId, dto.description, domainId, dto.icon, dto.name))
+        ResourceCreationDTO creationDTO = creationDTOJsonExtractor.parse(req.body());
+
+        Mono<OpenPaaSId> creatorIdMono = retrieveExistingUser(Username.of(creationDTO.creator))
+            .map(OpenPaaSUser::id);
+
+        Mono<OpenPaaSId> domainIdMono = domainDAO.retrieve(Domain.of(creationDTO.domain))
+            .switchIfEmpty(Mono.error(() -> new IllegalArgumentException(String.format("domain '%s' must exist", creationDTO.domain))))
+            .map(OpenPaaSDomain::id);
+
+        Mono<Map<Username, ResourceAdministrator>> adminMapMono = resolveAdministratorsFromDTO(creationDTO.administrators);
+
+        return Mono.zip(creatorIdMono, domainIdMono, adminMapMono)
+            .flatMap(tuple -> {
+                OpenPaaSId creatorId = tuple.getT1();
+                OpenPaaSId domainId = tuple.getT2();
+                Map<Username, ResourceAdministrator> adminMap = tuple.getT3();
+                return resourceDAO.insert(buildInsertRequest(adminMap, creatorId, creationDTO, domainId))
+                    .flatMap(resourceId -> resourceAdministratorService.setAdmins(domainId, resourceId, adminMap.keySet().stream().toList())
```

Comment:
```suggestion
                    .flatMap(resourceId -> resourceAdministratorService.setAdmins(domainId, resourceId, adminMap.keySet())
```

And use `Collection` instead of `List` in the base method